03_launch_mgmt_cluster.sh

```diff
@@ -79,6 +82,38 @@ function clone_repos() {
       git pull -r || true
       popd
     fi
+    #TODO Consider option to download prebaked clusterctl binary
+    if [[ -d "${CAPIPATH}" && "${FORCE_REPO_UPDATE}" == "true" ]]; then
+      rm -rf "${CAPIPATH}"
+    fi
+    if [ ! -d "${CAPIPATH}" ] ; then
+      pushd "${M3PATH}"
+      git clone "${CAPIREPO}" "${CAPIPATH}"
+      popd
+      pushd "${CAPIPATH}"
+      git checkout "${CAPIBRANCH}"
+      git pull -r || true
+      popd
+    fi
+
+}
+
+function patch_clusterctl(){
+  mkdir -p "${HOME}"/.cluster-api/overrides/infrastructure-metal3/"${CAPM3_RELEASE}"
+  cd "${HOME}"/.cluster-api/overrides/infrastructure-metal3/"${CAPM3_RELEASE}"
+  cat <<EOF> "kustomization.yaml"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- components.yaml
+EOF
+  wget https://github.com/metal3-io/cluster-api-provider-metal3/releases/download/"${CAPM3_RELEASE}"/cluster-template.yaml
+  wget https://github.com/metal3-io/cluster-api-provider-metal3/releases/download/"${CAPM3_RELEASE}"/infrastructure-components.yaml -O components.yaml
+  wget https://github.com/metal3-io/cluster-api-provider-metal3/releases/download/"${CAPM3_RELEASE}"/metadata.yaml
+  kustomize edit set image quay.io/metal3-io/cluster-api-provider-metal3:"${CAPM3_RELEASE}"=192.168.111.1:5000/localimages/cluster-api-provider-metal3:"${CAPM3_RELEASE}"
+  kustomize build . > infrastructure-components.yaml
+  rm components.yaml kustomization.yaml
```

Comment:
you should not be using the released version of capm3. The goal of those tests are to test a branch. Instead of doing it this way, you should generate the release manifests by doing `make release-manifests` . Then copy what is in the `out` folder into the override folder of clusterctl.

For the image, you can override it in the clusterctl config file, you don't need an intermediary kustomization. And the image should be overridden conditionally

Reply:
Mostly fixed, only image overwrite is a problem

Reply:
Completely fixed